src/wallet/wallet2.cpp

```diff
@@ -10324,6 +10323,58 @@ std::vector<std::pair<uint64_t, uint64_t>> wallet2::estimate_backlog(uint64_t mi
   return estimate_backlog(fee_levels);
 }
 //----------------------------------------------------------------------------------------------------
+uint64_t wallet2::get_segregation_fork_height() const
+{
+  if (m_nettype == TESTNET)
+    return TESTNET_SEGREGATION_FORK_HEIGHT;
+  if (m_nettype == STAGENET)
+    return STAGENET_SEGREGATION_FORK_HEIGHT;
+  THROW_WALLET_EXCEPTION_IF(m_nettype != MAINNET, tools::error::wallet_internal_error, "Invalid network type");
+
+  if (m_segregation_height > 0)
+    return m_segregation_height;
+
+  static const bool use_dns = true;
+  if (use_dns)
```

Comment:
Does this mean that those who don't want to use the DNS have to compile the code themselves?

Reply:
The user setting takes precedence

Reply:
I don't see the reason for the existence of this `if (use_dns) { ... }`.

Reply:
This is because there will likely be a --no-dns flag

Reply:
Ah, OK.